File: bhampick/scripts/client2.py
import rospy
from bhampick.srv import posedim, posedimRequest
from geometry_msgs.msg import PoseStamped
from tf.transformations import quaternion_from_euler

rospy.init_node('placecaller')

client = rospy.ServiceProxy('placeit', posedim)
srv = posedimRequest()

srv.posercam.header.frame_id = "base_link"
srv.posercam.header.stamp = rospy.Time.now()
srv.posercam.pose.position.x = 0.6
srv.posercam.pose.position.y = 0
srv.posercam.pose.position.z = 0.95

roll = 0
pitch = 0
yaw = 0
quat = quaternion_from_euler(roll, pitch, yaw)

srv.posercam.pose.orientation.x = quat[0]
srv.posercam.pose.orientation.y = quat[1]
srv.posercam.pose.orientation.z = quat[2]
srv.posercam.pose.orientation.w = quat[3]

srv.dimension.x = 0.0727
srv.dimension.y = 0.0727
srv.dimension.z = 0.4

try:
    rospy.wait_for_service('placeit')
    response = client(srv)
    rospy.loginfo("Pick Success: %s", response.graspsuccess)
except rospy.ServiceException as e:
    rospy.logerr("Failed to call service: %s", str(e))
    exit(1)

File: bhampick/src/client2.cpp
#include "ros/ros.h"
#include <bhampick/posedim.h>
#include <tf/transform_datatypes.h>

int main(int argc, char **argv)
{
  ros::init(argc, argv, "placecaller");

  ros::NodeHandle n;
  ros::ServiceClient client = n.serviceClient<bhampick::posedim>("placeit");
  bhampick::posedim srv;
  srv.request.posercam.header.frame_id = "base_link";
  srv.request.posercam.header.stamp = ros::Time::now();
  srv.request.posercam.pose.position.x = 0.8;
  srv.request.posercam.pose.position.y = 0;
  srv.request.posercam.pose.position.z = 0.9;

  double roll = 0;
  double pitch = 0;
  double yaw = 0;
  tf::Quaternion q;
  q.setRPY(roll, pitch, yaw);

  srv.request.posercam.pose.orientation.x = q[0];
  srv.request.posercam.pose.orientation.y = q[1];
  srv.request.posercam.pose.orientation.z = q[2];
  srv.request.posercam.pose.orientation.w = q[3];

  srv.request.dimension.x = 0.0264;
  srv.request.dimension.y = 0.0727;
  srv.request.dimension.z = 0.1363;

  if (client.call(srv))
  {
    ROS_INFO("Place Success : %s", srv.response.graspsuccess ? "true" : "false");
  }
  else
  {
    ROS_ERROR("Failed to call service");
    return 1;
  }

  return 0;
}